### PR TITLE
add default terms and refractor nda input

### PR DESCRIPTION
**Pull request by CDharmateja**
#1044

### CODE REVIEW

File: src/components/ChallengeEditor/ChallengeView/index.js [hunk 1]
```diff
@@ -179,11 +180,26 @@ const ChallengeView = ({
...
+                {params.get('beta') === 'true' && (
+                  <div className={styles.row}>
+                    <div className={styles.col}>
+                      <span>
+                        <span className={styles.fieldTitle}>Terms:</span>
+                        {challenge.terms.map(term => term.id).join(', ')}
+                      </span>
+                    </div>
+                  </div>
+                )}
```
Comment (maxceem) [inline src/components/ChallengeEditor/ChallengeView/index.js:195]: no need to show all the terms

File: src/components/ChallengeEditor/index.js [hunk 2]
```diff
@@ -1392,8 +1394,15 @@ class ChallengeEditor extends Component {
...
-                {false && (<TermsField terms={metadata.challengeTerms} challenge={challenge} onUpdateMultiSelect={this.onUpdateMultiSelect} />)}
+                <NDAField challenge={challenge} toggleNdaRequire={this.toggleNdaRequire} />
+                {params.get('beta') && (
+                  <TermsField
+                    terms={metadata.challengeTerms}
+                    projectTerms={projectDetail.terms}
+                    challenge={challenge}
+                    onUpdateMultiSelect={this.onUpdateMultiSelect}
+                  />
+                )}
```
Comment (maxceem) [inline src/components/ChallengeEditor/index.js:1405]: no need to edit al the terms, please, hide it using `flase` as it was before

**Review by maxceem — CHANGES_REQUESTED** [review]
@CDharmateja there are some missing requrements, let me clarify them:

1. We should not show all the terms and should be allowed to edit all the terms. Only enable/disable NDA. So no need to show terms and we should hide the editor for all the terms.

     image

      image
      
      - As we don't need to show Terms selector, please, cancel all the changes in the file `src/components/ChallengeEditor/Terms-Field/index.js` for now.

2. There is one missing requirement:
     > When we are creating a new challenge, we should apply all the terms of the projects which are listed in the project.terms array to the challenge. So new challenge has to be created with the same terms as project.
     
     So when we create a challenge now, we add only one hardcoded term https://github.com/topcoder-platform/work-manager/blob/develop/src/components/ChallengeEditor/index.js#L838. We should still add this term. But also, we should add all the terms of project, which are stored in `project.terms`. Each term should be added with the same roleId `SUBMITTER_ROLE_UUID`.

3. In the VIEW mode, let's show only text "yes" or "no" without radioboxes, like we do for other data 
   image

4. Can we please make smaller distance between yes and no options 

     image

**Comment by maxceem** [comment]
Also, could you please, resolve merge conflicts with the other PR which I've already merged.

File: src/components/ChallengeEditor/NDAField/index.js [hunk 3]
```diff
@@ -1,26 +1,58 @@
...
+
+  if (readOnly) {
+    return (
+      <div className={styles.row}>
+        <div className={cn(styles.field, styles.col1, styles.fieldTitle)}>NDA Required :</div>
+        <div className={cn(styles.field, styles.col2)}>
+          {
+            isRequiredNda
+              ? <div>Yes</div>
+              : <div>No</div>
```
Comment (maxceem) [inline src/components/ChallengeEditor/NDAField/index.js:19]: @CDharmateja I guess we don't need to wrap each answer "yes" / "no" into `<div>` and can simplify this to:
```js
{isRequiredNda ? 'Yes' : 'No'}
```

File: src/components/ChallengeEditor/NDAField/index.js [hunk 4]
```diff
@@ -1,26 +1,58 @@
...
+          <input
+            name='nda'
+            type='radio'
+            id='nda-yes'
+            checked={isRequiredNda}
+            onChange={toggleNdaRequire}
+          />
+          <label className={styles['tc-RadioButton-label']} htmlFor='nda-yes'>
+            <div>yes</div>
+            <input type='hidden' />
```
Comment (maxceem) [inline src/components/ChallengeEditor/NDAField/index.js:40]: @CDharmateja do we really need this "hidden" input?

File: src/components/ChallengeEditor/NDAField/index.js [hunk 5]
```diff
@@ -1,26 +1,58 @@
...
+          <input
+            name='nda'
+            type='radio'
+            id='nda-no'
+            checked={!isRequiredNda}
+            onChange={toggleNdaRequire}
+          />
+          <label className={styles['tc-RadioButton-label']} htmlFor='nda-no'>
+            <div>No</div>
+            <input type='hidden' />
```
Comment (maxceem) [inline src/components/ChallengeEditor/NDAField/index.js:53]: @CDharmateja do we really need this "hidden" input?

**Review by maxceem — CHANGES_REQUESTED** [review]
Thanks for update, @CDharmateja. Works great now. Just a few minor improvements:

1. Label for this field should be "bold" as all other labels:

   image
   
Also, there are few code improvement suggestions, please, kindly check comments below.

File: src/components/ChallengeEditor/NDAField/index.js (hunk 4 above)
Comment (CDharmateja) [reply src/components/ChallengeEditor/NDAField/index.js:40]: I think its not necessary. The input radio structure is like that here https://github.com/topcoder-platform/work-manager/blob/6a0579aa2b10f3345c65ec573e785db75ba6ea50/src/components/ChallengeEditor/ReviewType-Field/index.js#L31. So I have used it.

**Review by maxceem — APPROVED** [review]
@CDharmateja perfect now, thanks.

**Pull request merged**